src/test/java/io/github/lottetreg/httpserver/ReaderTest.java

```diff
@@ -8,23 +8,51 @@
 
 import java.io.*;
 
+
 public class ReaderTest {
   @Rule
   public ExpectedException exceptionRule = ExpectedException.none();
 
   @Test
-  public void itReadsFromTheConnection() {
+  public void readReturnsAnHTTPRequest() {
+    class MockConnection extends BaseMockConnection {
+      @Override
+      public InputStream getInputStream() {
+        String request =
+                "GET / HTTP/1.0\r\n" +
+                "Content-Length: 17\r\n" +
+                "\r\n" +
+                "some body to love";
```

Comment:
I have a feeling you will be writing many more of these request strings. An option would be create a test helper to build these requests for you so you could do something like:

```
requestBuilder
  .setMethod("GET")
  .setPath("/somePath")
  .setHeader("Content-", "text/plain")
 /// etc
  .build()
```
It's cleaner to read and It also allows you to harcode things like the CRLF (`\r\n') and the HTTP version in the builder.

Reply:
Good idea! Thanks, Kara!